scripts/generate_multilevel_sensor_constants.py

```diff
@@ -90,69 +107,142 @@ def normalize_scale_definition(scale_definitions: dict[str, dict]) -> dict[str,
+]
+
+lines.extend(
+    generate_int_enum_class_definition(
         "MultilevelSensorType",
         sensors,
         SENSOR_TYPE_URL,
         get_id_func=lambda x: x["id"],
     )
 )
 
+unit_name_to_enum_map = defaultdict(list)
 for scale_name, scale_dict in scales.items():
-    print(
-        generate_intenum_class_definition(
-            f"{format_for_class_name(scale_name)}Scale",
+    lines.extend(
+        generate_int_enum_class_definition(
+            format_for_class_name(scale_name),
             scale_dict,
             SENSOR_TYPE_URL,
             docstring_info=f"scales for {scale_name}",
         )
     )
+    for unit_name in scale_dict.keys():
+        unit_name_to_enum_map[unit_name].append(
+            f"{format_for_class_name(scale_name)}.{unit_name}"
+        )
+unit_name_to_enum_map = dict(
+    sorted(unit_name_to_enum_map.items(), key=lambda kv: kv[0])
+)
+for unit_name, enum_list in unit_name_to_enum_map.items():
+    unit_name_to_enum_map[unit_name] = sorted(enum_list)
 
-print("MULTILEVEL_SENSOR_TYPE_TO_SCALE_MAP = {")
+scale_class_names = [format_for_class_name(scale_name) for scale_name in scales]
+lines.extend(
+    [f"MultilevelSensorScaleType = Union[{', '.join(sorted(scale_class_names))}]", ""]
+)
+
+multilevel_sensor_type_to_scale_map_line: str = (
+    "MULTILEVEL_SENSOR_TYPE_TO_SCALE_MAP: Dict[MultilevelSensorType, "
+    "Type[MultilevelSensorScaleType]] = {"
+)
 for sensor_name, sensor_def in sensors.items():
-    print(
-        f"    MultilevelSensorType.{sensor_name}: {format_for_class_name(sensor_def['scale'])}Scale,"
+    multilevel_sensor_type_to_scale_map_line += (
+        f"    MultilevelSensorType.{sensor_name}: "
+        f"{format_for_class_name(sensor_def['scale'])},"
     )
-print("}")
-print()
-print()
+multilevel_sensor_type_to_scale_map_line += "}"
+lines.append(multilevel_sensor_type_to_scale_map_line)
+lines.append("")
+
+for unit_name, unit_enums in unit_name_to_enum_map.items():
+    lines.append(f"UNIT_{unit_name} = {{{','.join(sorted(unit_enums))}}}")
+
+lines.extend(
+    [
+        "",
+        "# ----------------------------------------------------------------------------------- #",
+        "# **END OF AUTOGENERATED CONTENT** (DO NOT EDIT/REMOVE THIS COMMENT BLOCK AND DO NOT  #",
+        "# EDIT ANYTHING ABOVE IT. IF A NEW IMPORT IS NEEDED, UPDATE THE LINES AROUND 135      #",
+        "# IN scripts/generate_multilevel_sensor_constants.py THEN RE-RUN THE SCRIPT. ALL      #",
+        "# LINES WRITTEN BELOW THIS BLOCK WILL BE PRESERVED AS LONG AS THIS BLOCK REMAINS)     #",
+        "# ----------------------------------------------------------------------------------- #",
+        "",
+    ]
+)
 
-scale_class_names = [
-    f"{format_for_class_name(scale_name)}Scale" for scale_name in scales
-]
-print(f"MultilevelSensorScaleType = Union[{', '.join(sorted(scale_class_names))}]")
-print()
-print()
+with open(MULTILEVEL_SENSOR_CONST_FILE_PATH, "r", encoding="utf-8") as fp:
```

Comment:
This can probably make it one line: https://docs.python.org/3/library/pathlib.html#pathlib.Path.read_text